pkg/sql/logictest/testdata/logic_test/fk

```diff
@@ -655,10 +655,10 @@ ALTER TABLE delivery DROP COLUMN "item"
 statement ok
 DROP INDEX products@products_upc_key CASCADE
 
-statement error "products" is referenced by foreign key from table "orders"
+statement error "products" is referenced by foreign key from table "customer reviews"
```

Comment:
I checked this and all the other test definition changes, they're legit. This commit messes with the ordering with which the undropped back-references are looked up, which in turn messes with the error messages.